src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,334 @@
+         * In Safari on iPhone and iPad, the window size is set to the size of
+         * the screen (minus Safari user interface controls), and cannot be changed
+         * by the user. To move around a webpage, the user changes the zoom level and position
+         * of the viewport as they double tap or pinch to zoom in or out, or by touching
+         * and dragging to pan the page. As a user changes the zoom level and position of the
+         * viewport they are doing so within a viewable content area of fixed size
+         * (that is, the window). This means that webpage elements that have their position
+         * "fixed" to the viewport can end up outside the viewable content area, offscreen.
+         *
+         * So the 'position: fixed' does not work on iPhone and iPad. To make it work,
+         * I need to use translate3d(0,0,0) to force Safari rerendering the sticky element.
+         **/
+        this.elem.style.transform = 'translate3d(0,0,0)';
+
+        this.elem.style.zIndex = this.zIndex;
+        this.elem.style.position = 'fixed';
+        this.elem.style.top = this.getCssNumber(this.upperScrollableContainer, 'top') + 'px';
+
+        this.scrollingRight = this.upperScrollableContainer.offsetLeft + this.upperScrollableContainer.offsetWidth;
+        let stuckRight: any = this.upperScrollableContainer.getBoundingClientRect().right;
+        this.elem.style.right = stuckRight + 'px';
+
+        this.elem.style.left = this.upperScrollableContainer.offsetLeft + 'px';
+        this.elem.style.bottom = 'auto';
+        this.elem.style.width = this.scrollingWidth + 'px';
+
+        // Set style for sticky element again for Mobile Views.
+        this.elem.style.setProperty('zIndex', this.zIndex);
+        this.elem.style.setProperty('position', 'fixed');
+        this.elem.style.setProperty('top', this.upperScrollableContainer.offsetTop + 'px');
+        this.elem.style.setProperty('right', stuckRight + 'px');
+        this.elem.style.setProperty('left', this.upperScrollableContainer.offsetLeft + 'px');
+        this.elem.style.setProperty('width', this.scrollingWidth + 'px');
+
+        this.activated.next(this.elem);
+    }
+
+    // unstuck element
+    unstickElement(): void {
+        this.isStuck = false;
+
+        this.elem.classList.add(this.stickEndClass);
+
+        this.stickyParent.style.position = 'relative';
+        this.elem.style.position = 'absolute';
+        this.elem.style.top = 'auto';
+        this.elem.style.right = 0;
+        this.elem.style.left = 'auto';
+        this.elem.style.bottom = 0;
+        this.elem.style.width = this.width;
+
+        this.deactivated.next(this.elem);
+    }
+
+
+    sticker(): void {
+        // detecting when a container's height changes
+        let currentContainerHeight: number = this.getCssNumber(this.stickyParent, 'height');
+        if (currentContainerHeight !== this.containerHeight) {
+            this.defineRestrictions();
```

Comment:
Usually `defineRestrictions()` are called before `sticker()`. Do you still need this?